dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewManagerScopeTest.kt

```diff
@@ -663,52 +636,40 @@ internal class RumViewManagerScopeTest {
-        forge: Forge,
-        @IntForgery(min = Build.VERSION_CODES.KITKAT, max = Build.VERSION_CODES.N) apiVersion: Int
+    fun `𝕄 send ApplicationStarted event 𝕎 handleEvent ()`(
+        forge: Forge
     ) {
         // Given
-        CoreFeature.processImportance = RunningAppProcessInfo.IMPORTANCE_FOREGROUND
-        whenever(mockBuildSdkVersionProvider.version()) doReturn apiVersion
-        val childView: RumViewScope = mock()
-        val startViewEvent = forge.startViewEvent()
-
-        // When
-        testedScope.onViewDisplayed(startViewEvent, childView, mockWriter)
-
-        // Then
-        argumentCaptor<RumRawEvent> {
-            verify(childView).handleEvent(capture(), same(mockWriter))
-
-            val event = firstValue as RumRawEvent.ApplicationStarted
-            assertThat(event.applicationStartupNanos).isEqualTo(RumFeature.startupTimeNs)
+        // Because we have to test that the `application_started` action is written, we need to set
+        // up the feature scope properly
+        val mockRumFeatureScope: FeatureScope = mock()
+        val mockEventBatchWriter: EventBatchWriter = mock()
+        whenever(mockSdkCore.getFeature(RumFeature.RUM_FEATURE_NAME)) doReturn mockRumFeatureScope
+        whenever(mockRumFeatureScope.withWriteContext(any(), any())) doAnswer {
+            val callback = it.getArgument<(DatadogContext, EventBatchWriter) -> Unit>(1)
+            callback.invoke(fakeDatadogContext, mockEventBatchWriter)
         }
-        verifyZeroInteractions(mockWriter)
-    }
-
-    @Test
-    fun `𝕄 send ApplicationStarted event 𝕎 onViewDisplayed() {Nougat+}`(
-        forge: Forge,
-        @IntForgery(min = Build.VERSION_CODES.N) apiVersion: Int
-    ) {
-        // Given
+        val appStartTimeNs = System.nanoTime() - TimeUnit.MILLISECONDS.toNanos(500)
+        whenever(mockAppStartTimeProvider.appStartTimeNs) doReturn appStartTimeNs
         CoreFeature.processImportance = RunningAppProcessInfo.IMPORTANCE_FOREGROUND
-        whenever(mockBuildSdkVersionProvider.version()) doReturn apiVersion
-        val childView: RumViewScope = mock()
-        val startViewEvent = forge.startViewEvent()
+
+        val fakeEvent = forge.addErrorEvent()
 
         // When
-        testedScope.onViewDisplayed(startViewEvent, childView, mockWriter)
+        testedScope.handleEvent(fakeEvent, mockWriter)
 
         // Then
-        argumentCaptor<RumRawEvent> {
-            verify(childView).handleEvent(capture(), same(mockWriter))
-
-            val event = firstValue as RumRawEvent.ApplicationStarted
-            assertThat(event.applicationStartupNanos)
-                .isCloseTo(System.nanoTime(), Offset.offset(TimeUnit.MILLISECONDS.toNanos(100)))
+        val appStartTimeMs = TimeUnit.NANOSECONDS.toMillis(appStartTimeNs)
+        argumentCaptor<ActionEvent> {
+            verify(mockWriter, atLeastOnce()).write(eq(mockEventBatchWriter), capture())
```

Comment:
Do we expect more calls here? If we expect only one call, we can use `times(1)`, or simply omit it and use `verify(mockWriter).write(...)`

Reply:
Oh yeah, it makes sense since we create Application Launch scope and then delegate action to it.

And I guess then the number of interactions vary from the type of the original event handled? Because some events can cause an instant write (like fatal errors).